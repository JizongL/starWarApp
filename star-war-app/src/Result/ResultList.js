import React from 'react'
import starWarContext from '../context/starWarContext'

class ResultList extends React.Component{
  static contextType = starWarContext
  
  handleResult =()=>{
   if(this.props.data.results){ return this.props.data.results.map((result,index)=>{
      return <li key={index}>{result.name}</li>
    })}
  }

  render(){
    console.log(this.props.data.results,'test context')
    
    return(
      <div>
      <ul>
       {this.handleResult()}
      </ul>
      </div>
    )
  }
}

export default ResultList