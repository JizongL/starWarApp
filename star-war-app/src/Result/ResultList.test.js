import React from 'react'
import ReactDOM from 'react-dom'
import ResultList from './ResultList'

const object = {0:
  {birth_year: "19BBY",
  created: "2014-12-09T13:50:51.644000Z",
  edited: "2014-12-20T21:17:56.891000Z",
  eye_color: "blue",
  films:["https://swapi.co/api/films/2/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/", "https://swapi.co/api/films/7/"],
  gender: "male",
  hair_color: "blond",
  height: "172",
  homeworld: "https://swapi.co/api/planets/1/",
  mass: "77",
  name: "Luke Skywalker",
  skin_color: "fair"}}

it('renders without crashing',()=>{
  const div= document.createElement('div')
  ReactDOM.render(<ResultList data={object}/>,div)
  ReactDOM.unmountComponentAtNode(div)
})