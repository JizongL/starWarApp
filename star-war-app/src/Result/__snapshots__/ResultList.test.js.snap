// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redners the UI as expected 1`] = `[Function]`;

exports[`test ResultList redners the UI as expected 1`] = `[Function]`;
